### PR TITLE
test/integration/scheduler_perf: make sure each testCase and workload has a unique name

### LINKED ISSUE #93795: test/integration/scheduler_perf: make sure each testCase and workload has a unique name
Note: this issue should be viewed in the context of #93252.

**What would you like to be added**: When parsing the performance-config.yaml file, make sure each testCase and workload has a unique name. This can also be implemented by changing the yaml file's structure into maps instead of arrays (as suggested by @alculquicondor), but this will result in a loss of test/workload order (i.e. a `5000Nodes` workload may run before a `500Nodes` workload).

**Why is this needed**: The name of each testCase and workload is used in the report generated by the framework and should therefore be unique in order to identify each metric.

/sig scheduling
/sig testing
/assign

### PR DESCRIPTION
**Pull request by nicomitchell**
/kind feature
#### What this PR does / why we need it:
Adds checks inside of scheduler_perf_test.go to ensure test case and workload names in performance-config.yaml are unique. 

#### Which issue(s) this PR fixes:
Fixes #93795 

#### Special notes for your reviewer:
This is my first contribution so let me know if there's anything I could've done differently.
#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/ok-to-test

File: test/integration/scheduler_perf/scheduler_perf_test.go [hunk 1]
```diff
@@ -111,6 +111,17 @@ func (tc *testCase) collectsMetrics() bool {
 	return false
 }
 
+func (tc *testCase) workloadNamesUnique() bool {
+	workloadUniqueNames := map[string]bool{}
+	for _, w := range tc.Workloads {
+		if workloadUniqueNames[w.Name] {
+			return false
```
Comment (adtac) [inline test/integration/scheduler_perf/scheduler_perf_test.go:118]: nit: I would return an error with the conflicting workload name instead

**Review by adtac — COMMENTED** [review]
one minor nit, lgtm otherwise, thanks!

File: test/integration/scheduler_perf/scheduler_perf_test.go (hunk 1 above)
Comment (nicomitchell) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: Sure, done!

**Review by adtac — COMMENTED** [review]
/lgtm

**Comment by nicomitchell** [comment]
Pinging @ravisantoshgudimetla for approval when you get the chance

**Review by ravisantoshgudimetla — APPROVED** [review]
Thank you for working on this @nicomitchell 

/approve

**Pull request merged**